src/common_utils/language_utils/statement_utils.py

```diff
@@ -64,13 +72,19 @@ def extract_single_tags(single_or_complex_tag, separator=':'):
     return single_or_complex_tag.split(separator)
 
 
-
 def is_present_in_set(single_tag, set):
     for single_or_complex_tag in set:
         single_tags = extract_single_tags(single_or_complex_tag)
         for tag in single_tags:
             if single_tag == tag:
                 return True
-            elif single_tag in NOT_FULL_MEANINGFUL_SYNONYMS and tag in NOT_FULL_MEANINGFUL_SYNONYMS:
+            elif single_tag in UNIV_SYNONYMS and tag in UNIV_SYNONYMS:
                 return True
     return False
+
+
+def contains_synonym(words):
+    for word in words:
+        if word in UNIV_SYNONYMS:
+            return True
+    return False
```

Comment:
isn't it equivalent and more elegant solution? 
`def contains_synonym(words):
return any(word in words for word in UNIV_SYNONYMS) `